spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -44,6 +46,31 @@
     import gettext
     _ = gettext.gettext
 
+
+def _find_pylintrc_path(path):
+    os.chdir(path)
+    return pylint.config.find_pylintrc()
+
+
+def get_pylintrc_path(search_paths=None, home_path=None):
+    """Get the path to the highest pylintrc file on a set of search paths."""
+    current_cwd = os.getcwd()
+    pylintrc_path = None
+    if home_path is None:
+        home_path = osp.expanduser("~")
+    try:
+        pylintrc_paths = [
+            _find_pylintrc_path(path) for path in search_paths if path]
+        pylintrc_path_home = _find_pylintrc_path(home_path)
+        for pylintrc_path in pylintrc_paths:
+            if (pylintrc_path is not None
+                    and pylintrc_path != pylintrc_path_home):
+                break

```

Comment:
What is the part in this code that we are trying to catch?

Ideally try/catch should be as small as possible to be very granular about what we are catching.

Reply:
Its not excepting ("catching") anything, as if you look at the function this isn't a try-except at all, its a try-finally, in order to ensure the working directory gets set back to what it was prior to the function's execution. I normally include an explanatory comment documenting the purpose of any such similar construct, but I must not have added when I originally wrote the code since I didn't intend to implement it myself. I've added it now for the next commit.